core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -623,4 +626,23 @@ ReentrantReadWriteLock getLock(WorkerMetaLockSection lockType) {
   public LockResource lockWorkerMeta(EnumSet<WorkerMetaLockSection> lockTypes, boolean isShared) {
     return new LockResource(new WorkerMetaLock(lockTypes, isShared, this));
   }
+
+  /**
+   * Locks the corresponding read locks for the specified worker information fields.
+   * This is a wrapper of {@link MasterWorkerInfo#lockWorkerMeta(EnumSet, boolean)}
+   *
+   * @param fieldRange a set of {@link WorkerInfoField}
+   * @return a {@link LockResource} of the {@link WorkerMetaLock}
+   */
+  public LockResource lockWorkerMetaForInfo(
```

Comment:
This is just a wrapper to make the logic easier for user.

In terms of why we don't lock internally in `MasterWorkerInfo`, the methods in this class are usually called externally in combination so it's not easy for ALL methods here to embed locking internally. For consistency (so that we don't have some methods that lock internally and the other externally), we require ALL methods here to lock externally if needed.

Feel free to drop a comment if you have better ideas here.